x-pack/plugins/rule_registry/server/rule_data_client/rule_data_client.ts

```diff
@@ -131,34 +133,35 @@ export class RuleDataClient implements IRuleDataClient {
     };
   }
 
-  public getWriter(options: { namespace?: string } = {}): IRuleDataWriter {
+  public async getWriter(options: { namespace?: string } = {}): Promise<IRuleDataWriter> {
     const namespace = options.namespace || 'default';
     const cachedWriter = this.writerCache.get(namespace);
     const isWriterCacheEnabled = () => this.writerCacheEnabled;
 
     // There is no cached writer, so we'll install / update the namespace specific resources now.
     if (!isWriterCacheEnabled() || !cachedWriter) {
-      const writerForNamespace = this.initializeWriter(namespace);
+      const writerForNamespace = await this.initializeWriter(namespace);
       this.writerCache.set(namespace, writerForNamespace);
       return writerForNamespace;
     } else {
       return cachedWriter;
     }
   }
 
-  private initializeWriter(namespace: string): IRuleDataWriter {
+  private async initializeWriter(namespace: string): Promise<IRuleDataWriter> {
     const isWriteEnabled = () => this.writeEnabled;
     const turnOffWrite = () => (this.writeEnabled = false);
 
     const { indexInfo, resourceInstaller } = this.options;
     const alias = indexInfo.getPrimaryAlias(namespace);
 
     // Wait until both index and namespace level resources have been installed / updated.
-    const prepareForWriting = async () => {
-      if (!isWriteEnabled()) {
-        throw new RuleDataWriteDisabledError();
-      }
+    if (!isWriteEnabled()) {
+      this.options.logger.debug(`Writing is disabled, bulk() will not write any data.`);
+      throw new RuleDataWriteDisabledError();
```

Comment:
As mentioned in my previous comment, this error will incorrectly tell user to check "xpack.ruleRegistry.write.enabled" configuration in "kibana.yml" file when writer fails due to the index error

Reply:
Updated to differentiate between types of disabled in this commit: https://github.com/elastic/kibana/pull/134280/commits/dc17a5a9ce2589d4d216aadbe61c49e6e7df534b